src/main/java/seedu/duke/TaskList.java

```diff
@@ -0,0 +1,130 @@
+package main.java.seedu.duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+
+import main.java.seedu.duke.todo.Task;
+
+/**
+ * Represents a list of tasks. A task list supports various modifications of the task list.
+ */
+public class TaskList {
+    ArrayList<Task> tasks = new ArrayList<>();
+    private DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMM d yyyy");
+
+    /**
+     * Lists out all the tasks in the list.
+     * @throws DukeException
+     */
+    public void listTasks() throws DukeException {
+        if (tasks.size() <= 0) {
+            throw new DukeException("You don't have any tasks.");
+        }
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 0; i < tasks.size(); i++) {
+            int taskNo = i + 1;
+            Task task = tasks.get(i);
+            System.out.println(taskNo + "." + task);
+        }
+    }
+
+    /**
+     * Lists out all the tasks on a specific date.
+     * @param date The specific date.
+     * @throws DukeException If there is no task on that day.
+     */
+    public void listTasksOn(LocalDate date) throws DukeException {
+        if (tasks.size() <= 0) {
+            throw new DukeException("You don't have any tasks.");
+        }
+        ArrayList<Task> taskList = new ArrayList<>();
```

Comment:
I think the name of the taskList should be changed since it can be mistaken with tasks!